### PR TITLE
Make corporate redemption codes lower case 

### PR DESCRIPTION
**Pull request by rupertbates**
## Why are you doing this?
We currently have inconsistent casing conventions between gift and corporate codes, gift uses lowercase whereas corporate uses uppercase.

This PR changes corporate codes to use lowercase, I have also reintroduced the code length check when constructing a `RedemptionCode`.

### TODO: 

- [ ] Fix two integration tests which rely on values in Dynamo and Zuora

[**Trello Card**](https://trello.com/c/vuOwsdwq/3189-make-corporate-sub-codes-lowercase)

### CODE REVIEW

File: support-models/src/main/scala/com/gu/support/config/Stages.scala [hunk 1]
```diff
@@ -18,7 +18,7 @@ object Stages {
 object Stage {
   def fromString(s: String): Option[Stage] = condOpt(s) {
     case "DEV" => DEV
-    case "CODE" => CODE
+    case "code" => CODE
```
Comment (johnduffell) [inline support-models/src/main/scala/com/gu/support/config/Stages.scala:21]: 🤣

File: support-models/src/test/scala/com/gu/support/redemptions/RedemptionCodeSpec.scala [hunk 2]
```diff
@@ -8,14 +8,14 @@ class RedemptionCodeSpec extends AnyFlatSpec with Matchers {
...
-      RedemptionCode(s"CODE$char").isLeft should be(true)
+      RedemptionCode(s"code-123456-$char").isLeft should be(true)
     }
   }
 
   it should "allow valid chars" in {
     val codes = """-0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"""
     codes.foreach { char =>
-      RedemptionCode(s"CODE$char").isRight should be(true)
+      RedemptionCode(s"code-123456-$char").isRight should be(true)
```
Comment (johnduffell) [inline support-models/src/test/scala/com/gu/support/redemptions/RedemptionCodeSpec.scala:18]: pull out the prefix into a shared variable if you like?

File: support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala [hunk 3]
```diff
@@ -13,91 +13,91 @@ class CorporateCodeValidatorSpec extends AsyncFlatSpec with Matchers {
...
+      codeValidator.getStatus(RedemptionCode("test-code-123").right.get)
     }
   }
 
   it should "handle a missing attribute code - available" in {
     val codeValidator = CorporateCodeValidator.withDynamoLookup {
-      case "CODE" => Future.successful(Some(Map(
-        "ASDFNQWEOIDNSDKNFNAKNDAKNSKANSDKNASDAKSND" -> DynamoBoolean(true),
+      case "test-code-123" => Future.successful(Some(Map(
+        "asdfnqweoidnsdknfnakndaknskansdknasdaksnd" -> DynamoBoolean(true),
```
Comment (johnduffell) [inline support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala:74]: this is not a code, it's just a random attribute name in place of `"available"`

Comment (johnduffell) [reply support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala:74]: consider calling it "arbitraryStrangeAttributeName" or something

File: support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala [hunk 4]
```diff
@@ -13,91 +13,91 @@ class CorporateCodeValidatorSpec extends AsyncFlatSpec with Matchers {
...
+      codeValidator.getStatus(RedemptionCode("test-code-123").right.get)
     }
   }
 
   it should "handle a missing attribute code - corporate id" in {
     val codeValidator = CorporateCodeValidator.withDynamoLookup {
-      case "CODE" => Future.successful(Some(Map(
-        "ASDFNQWEOIDNSDKNFNAKNDAKNSKANSDKNASDAKSND" -> DynamoString("1"),
+      case "test-code-123" => Future.successful(Some(Map(
+        "asdfnqweoidnsdknfnakndaknskansdknasdaksnd" -> DynamoString("1"),
```
Comment (johnduffell) [inline support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala:86]: same for this one, this is just an arbitrary attrubte in place of corporate id

File: support-services/src/test/scala/com/gu/support/redemption/corporate/SetCodeStatusITSpec.scala [hunk 5]
```diff
@@ -18,7 +18,7 @@ class SetCodeStatusITSpec extends AsyncFlatSpec with Matchers {
 
   // this is one test because it depends on external state which may not be in a particular state
   "setCodeStatus" should "set a code available and used" in {
-    val mutableCode: RedemptionCode = RedemptionCode("ITTEST-MUTABLE").right.get
+    val mutableCode: RedemptionCode = RedemptionCode("ittest-mutable").right.get
```
Comment (johnduffell) [inline support-services/src/test/scala/com/gu/support/redemption/corporate/SetCodeStatusITSpec.scala:21]: will need to update this script and clean the DEV dyamo table and rereun the script 
https://github.com/guardian/support-frontend/blob/8e9954cb2b2c11345cfe4ff778da972b99e03702/support-services/src/test/scala/com/gu/support/redemption/corporate/IT_testdata.sh#L16-L40

File: support-workers/src/test/scala/com/gu/support/workers/integration/CreateZuoraSubscriptionSpec.scala [hunk 6]
```diff
@@ -50,7 +50,7 @@ class CreateZuoraSubscriptionSpec extends AsyncLambdaSpec with MockServicesCreat
     val dynamoTableAsync: DynamoTableAsync = RedemptionTable.forEnvAsync(TouchPointEnvironments.SANDBOX)
     val setCodeStatus = SetCodeStatus.withDynamoLookup(dynamoTableAsync)
     val codeValidator = CorporateCodeValidator.withDynamoLookup(dynamoTableAsync)
-    val mutableCode: RedemptionCode = RedemptionCode("ITTEST-MUTABLE").right.get
+    val mutableCode: RedemptionCode = RedemptionCode("ITTEST-MUTABLE").right.get //TODO: RB make this work
```
Comment (johnduffell) [inline support-workers/src/test/scala/com/gu/support/workers/integration/CreateZuoraSubscriptionSpec.scala:53]: see the above script to make it work

File: support-models/src/main/scala/com/gu/support/config/Stages.scala (hunk 1 above)
Comment (rupertbates) [reply support-models/src/main/scala/com/gu/support/config/Stages.scala:21]: Oops, I was looking out for that as well!

File: support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala (hunk 3 above)
Comment (rupertbates) [reply support-services/src/test/scala/com/gu/support/redemption/corporate/CorporateCodeValidatorSpec.scala:74]: Ah well, might as well go lower case!

**Review by johnduffell — APPROVED** [review]
looks good, thanks for picking that one up! 👍

**Comment by johnduffell** [comment]
merge conflict that git couldn't detect https://github.com/guardian/support-frontend/blame/b0030a0e0e7263b666479850006959913e982fa0/support-frontend/test/utils/CheckoutValidationRulesTest.scala#L163

**Pull request merged**